index.js

```diff
@@ -174,13 +174,8 @@ function compression (options) {
       }
 
       // compression method
-      var accept = accepts(req)
-      var method = accept.encoding(['gzip', 'deflate', 'identity'])
-
-      // we really don't prefer deflate
-      if (method === 'deflate' && accept.encoding(['gzip'])) {
-        method = accept.encoding(['gzip', 'identity'])
-      }
+      var negotiator = new Negotiator(req)
+      var method = negotiator.encoding(['gzip', 'deflate', 'identity'], ['gzip'])
```

Comment:
When you add brotli, just add it here:

```suggestion
      var method = negotiator.encoding(['gzip', 'deflate', 'identity'], hasBrotli ? ['br', 'gzip'] : ['gzip'])
```